From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Greg Price <greg@zulip.com>
Date: Thu, 16 Jun 2022 21:13:31 -0700
Subject: [tsflower] rnav: Handle TS reference pragma for React
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

This seems like not a great practice in the library, but TsFlower
should deal with it anyway.

What's happening is that in the input file, instead of importing React
in a normal JS way, it's using a TypeScript `<reference … />` comment
to bring into scope everything defined in `@types/react`, including the
`React` namespace.

As a result, TsFlower currently isn't able to spot that the references
to a `React` namespace are referring to the 'react' module, and doesn't
rewrite references like `React.ReactNode`.
---
 .../elements/lib/typescript/src/types.js.flow         | 11 ++++++-----
 1 file changed, 6 insertions(+), 5 deletions(-)

diff --git types/@react-navigation/elements/lib/typescript/src/types.js.flow types/@react-navigation/elements/lib/typescript/src/types.js.flow
index 1270d287b..b28e5b4eb 100644
--- types/@react-navigation/elements/lib/typescript/src/types.js.flow
+++ types/@react-navigation/elements/lib/typescript/src/types.js.flow
@@ -1,6 +1,7 @@
 /* @flow
  * @generated by TsFlower
  */
+import type { ReactNode } from 'tsflower/subst/react';
 import type {
   WithAnimatedValue as $tsflower_subst$RN$Animated$WithAnimatedValue,
   StyleProp as $tsflower_subst$RN$StyleProp,
@@ -18,7 +19,7 @@ export type Layout = {
 };
 
 export type HeaderOptions = {
-  headerTitle?: string | ((props: HeaderTitleProps) => React.ReactNode),
+  headerTitle?: string | ((props: HeaderTitleProps) => ReactNode),
   headerTitleAlign?: 'left' | 'center',
   headerTitleStyle?: $tsflower_subst$RN$Animated$WithAnimatedValue<
     $tsflower_subst$RN$StyleProp<$tsflower_subst$RN$TextStyle>,
@@ -33,29 +34,29 @@ export type HeaderOptions = {
     pressOpacity?: number,
     labelVisible?: boolean,
     ...
-  }) => React.ReactNode,
+  }) => ReactNode,
   headerLeftLabelVisible?: boolean,
   headerLeftContainerStyle?: $tsflower_subst$RN$Animated$WithAnimatedValue<
     $tsflower_subst$RN$StyleProp<$tsflower_subst$RN$ViewStyle>,
   >,
   headerRight?: (props: {
     tintColor?: string,
     pressColor?: string,
     pressOpacity?: number,
     ...
-  }) => React.ReactNode,
+  }) => ReactNode,
   headerRightContainerStyle?: $tsflower_subst$RN$Animated$WithAnimatedValue<
     $tsflower_subst$RN$StyleProp<$tsflower_subst$RN$ViewStyle>,
   >,
   headerPressColor?: string,
   headerPressOpacity?: number,
   headerTintColor?: string,
   headerBackground?: (props: {
     style: $tsflower_subst$RN$Animated$WithAnimatedValue<
       $tsflower_subst$RN$StyleProp<$tsflower_subst$RN$ViewStyle>,
     >,
     ...
-  }) => React.ReactNode,
+  }) => ReactNode,
   headerBackgroundContainerStyle?: $tsflower_subst$RN$Animated$WithAnimatedValue<
     $tsflower_subst$RN$StyleProp<$tsflower_subst$RN$ViewStyle>,
   >,
@@ -90,7 +91,7 @@ export type HeaderButtonProps = {
 export type HeaderBackButtonProps = HeaderButtonProps & {
   disabled?: boolean,
   onPress?: () => void,
-  backImage?: (props: { tintColor: string, ... }) => React.ReactNode,
+  backImage?: (props: { tintColor: string, ... }) => ReactNode,
   label?: string,
   truncatedLabel?: string,
   labelVisible?: boolean,
-- 
2.32.0

